api/src/main/java/org/eclipse/microprofile/config/inject/ConfigProperty.java

```diff
@@ -100,6 +104,8 @@
 @Qualifier
 @Retention(RUNTIME)
 @Target({METHOD, FIELD, PARAMETER, TYPE})
+@Requirement(namespace = CDI_EXTENSION_PROPERTY, name = "org.eclipse.microprofile.config", effective = "active")
+@Requirement(namespace = CDI_EXTENSION_PROPERTY, name = "org.eclipse.microprofile.config", resolution = OPTIONAL)
```

Comment:
This is very fishy. Why do you have to add `Requirement` twice? Why cannot it cope with both effective and resolution?

Reply:
@bjhargrave you are exactly correct!

BTW, in times past when I've discussed this with @pkriens and @tjwatson we've dubbed this combination of two requirements, one optional, one effective active as a _weak requirement_.

A _weak requirement_ is used because IBM clearly does not want to strictly require Service Loader Mediator. This avoids that. Meanwhile, anyone who does want to use the resolver for assembly can simply enable `active` effectiveness.

About the OSGi CDI extension _name_, if folks are in agreement we could document that in the spec.

Reply:
> A _weak requirement_ is used because IBM clearly does not want to strictly require Service Loader Mediator.

The case here is not about SLM. It is about the fact that there are some CDI extension implementations which do not provide the capability for resolve time.

Reply:
Well, it's the exact same reason. If someone doesn't want to use OSGi CDI and manually wire things up it will still work.

Reply:
So this PR should put comments in the source code where we apply the `weak requirements` to document why there are 2 seemingly duplicate requirements and why we use `effective:=active` for non-runtime resolution scenarios.

Reply:
That sounds reasonable! I'll do that.

Reply:
That was my original objection for this PR with the requirement for runtime to provide CDI capability. Otherwise the bundle does not resolve. I think with this weak requirement, it is reasonable change. Agreed with @bjhargrave to add comments to explain this.

Reply:
We also need to agree on, and document in the spec, the name of the CDI extension for Config. @rotty3000 proposes `org.eclipse.microprofile.config` which seems fine to me. But others (spec readers and implementors) will need to know the agreed upon name.